### PR TITLE
JWT Auth implemented

### PR DESCRIPTION
**Pull request by tamurtaj85**
Here is the PR for complete JWT Auth implementation. Please review it and leave the remarks. Thanks

### CODE REVIEW

File: api/configs/configs.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+  secrets: {
+    jwt: process.env.JWT_SECRET || "nodejsAPI",
+    jwtExp: "100d",
+  },
+
+  dbUrl: "mongodb://localhost/ToDo_db",
+};
+
+export const constants = {
```
Comment (bilalk1) [inline api/configs/configs.js:12]: This file name should be index.js

File: api/configs/express.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import express from "express";
+import bodyParser from "body-parser";
+import { config } from "./configs.js";
+import { routes_ToDoList, routes_User, routes_Auth } from "../routes/index.js";
```
Comment (bilalk1) [inline api/configs/express.js:4]: Replace it with `import { routes_ToDoList, routes_User, routes_Auth } from "../routes";`

File: api/configs/express.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import { notFound, protect } from "../middleWare/middleware.js";
+
```
Comment (bilalk1) [inline api/configs/express.js:6]: please replace it with `import { notFound, protect } from "../middleWare";`

File: api/configs/express.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+const app = express();
+
+app.use(bodyParser.urlencoded({ extended: true }));
+app.use(bodyParser.json());
+
+await routes_ToDoList(app);
```
Comment (bilalk1) [inline api/configs/express.js:12]: why are you using await key word here

File: api/configs/mongooes.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import mongoose from "mongoose";
+import { config } from "./configs.js";
+
+// mongoose instance connection url connection
+mongoose.Promise = global.Promise;
+mongoose.connect(config.dbUrl);
+
+await import("../models/index.js"); // Loading all the residing models from [model/index.js]
```
Comment (bilalk1) [inline api/configs/mongooes.js:8]: ("../models");

File: api/controllers/authController.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+import jwt from "jsonwebtoken";
+
+export function newToken(user) {
+  return jwt.sign({ id: user.id }, config.secrets.jwt, {
+    expiresIn: config.secrets.jwtExp,
+  });
+}
+
+export function verifyToken(token) {
+  return new Promise((resolve, reject) => {
```
Comment (bilalk1) [inline api/controllers/authController.js:12]: Why you wrap your call back function into  promises ?

File: api/controllers/toDoListController.js [hunk 7]
```diff
@@ -1,37 +1,39 @@
 "use strict";
 
-const mongoose = require("mongoose");
+import { Tasks } from "../models/index.js";
```
Comment (bilalk1) [inline api/controllers/toDoListController.js:3]: please replace it with `import { Tasks } from "../models";`

File: api/controllers/toDoListController.js [hunk 8]
```diff
@@ -1,37 +1,39 @@
...
+import { paginationResponse } from "../utils/pagination.js";
```
Comment (bilalk1) [inline api/controllers/toDoListController.js:4]: Please re[place it with `"../utils";`

File: api/middleWare/middleware.js [hunk 9]
```diff
@@ -0,0 +1,34 @@
+import { User } from "../models/index.js";
+import { authControl } from "../controllers/index.js";
```
Comment (bilalk1) [inline api/middleWare/middleware.js:2]: Update the name of the middleware to `auth.js`

File: package.json [hunk 10]
```diff
@@ -2,18 +2,20 @@
...
     "nodemon": "^2.0.13"
   },
   "dependencies": {
+    "bcrypt": "^5.0.1",
     "express": "^4.17.1",
+    "jsonwebtoken": "^8.5.1",
     "mongoose": "^6.0.8",
+    "mongoose-paginate-v2": "^1.4.2",
     "package.json": "^2.0.1"
   }
```
Comment (bilalk1) [inline package.json:20]: Do not use index while importing the file in this PR

File: api/configs/express.js (hunk 4 above)
Comment (tamurtaj85) [reply api/configs/express.js:12]: Because they all are async functions.

File: api/configs/configs.js (hunk 1 above)
Comment (tamurtaj85) [reply api/configs/configs.js:12]: There is already a file named index.js in configs folder

File: api/configs/express.js (hunk 2 above)
Comment (tamurtaj85) [reply api/configs/express.js:4]: This change gives the following error:
Error [ERR_UNSUPPORTED_DIR_IMPORT]: Directory import '/home/tamurtaj/Documents/OfficeWork/CRUDapi/api/routes/' is not supported resolving ES modules imported from /home/tamurtaj/Documents/OfficeWork/CRUDapi/api/configs/express.js

File: api/configs/express.js (hunk 3 above)
Comment (tamurtaj85) [reply api/configs/express.js:6]: Same error as earlier.

File: api/configs/mongooes.js (hunk 5 above)
Comment (tamurtaj85) [reply api/configs/mongooes.js:8]: Gives the same error as earlier.

File: api/controllers/toDoListController.js (hunk 7 above)
Comment (tamurtaj85) [reply api/controllers/toDoListController.js:3]: Gives the same error as earlier.

File: api/controllers/toDoListController.js (hunk 8 above)
Comment (tamurtaj85) [reply api/controllers/toDoListController.js:4]: Gives the same error as earlier.

**Pull request merged**